### PR TITLE
[processor/websocketprocessor] add websocket processor skeleton

### PR DESCRIPTION
**Pull request by pmcollins**
**Description:** This change adds just the config, factory, and initial README for the WebSocket processor.

**Link to tracking Issue:** #19633 

**Testing:** Unit tests added.

**Documentation:** README added.

### CODE REVIEW

File: processor/websocketprocessor/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Websocket Processor
```
Comment (atoulme) [inline processor/websocketprocessor/README.md:1]: we can add the status metadata table later in a separate PR.

**Review by atoulme — APPROVED** [review]

**Review by hughesjj — APPROVED** [review]

**Comment by pmcollins** [comment]
Hi @jpkrohling -- can this be merged?

File: processor/websocketprocessor/go.mod [hunk 2]
```diff
@@ -0,0 +1,42 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/processor/websocketprocessor
+
+go 1.19
+
+require (
+	github.com/stretchr/testify v1.8.2
+	go.opentelemetry.io/collector v0.75.0
+	go.opentelemetry.io/collector/component v0.75.0
```
Comment (atoulme) [inline processor/websocketprocessor/go.mod:8]: You are going to need to update versions here.

**Comment by atoulme** [comment]
Please rebase and update versions, sorry.

File: processor/websocketprocessor/go.mod (hunk 2 above)
Comment (pmcollins) [reply processor/websocketprocessor/go.mod]: Thanks. Should be done now.

**Comment by pmcollins** [comment]
Hi @atoulme -- since I'm wrapping up my work on the collector do you mind if I close this PR? This functionality can be added later if there is interest.

**Comment by atoulme** [comment]
I think if possible we can land it as is and go from there. I will follow up since I sponsored.

**Pull request merged**